### DISCUSSION TITLE
A Quick-checks sheet for instance admins.

### DISCUSSION THREAD

**Discussion by BernieHuang2008** [searxng/searxng#2896] [Ideas]
Instance admins may be confused by some Http-errors. Some are very simple, e.g. 403-forbidden( #2796 ) is caused by bot-attact.

Some admins just report these simple issues to us directly, it's really board to answer their issues.

As a result, I suggest that we could build a Quick-check-sheet that admins can find their problems easily.

For example, it could be like this:

|error type|error code|check steps|reason|solution|
|---|---|---|---|---|
|HTTP error|403|Use browser to check if you can visit the engine site manually, if you see `captcha`, ...|you are under bot-attack|Enable bot - protection in searxng and wait some days|

Like this, I think it will be more convenient both for admins and maintainers.

**Comment by BernieHuang2008** [comment]
Or, more Convienent, we can write a script that Auto-checks the problem and provide solution.

**Reply by return42** [reply]
Not sure what other think about, but I think having wiki for trouble  shoot and best practice would be good. 

.. meanings?

**Reply by BernieHuang2008** [reply]
I mean a program to do the wiki's job --- map the error code with the solution, so users won't need to read the wiki.

This program can be auto - activated when a server is dead (since it's just a mapping, it won't be expensive). This made me think of that `better error message` issue -- more precise informations can be provided.

**Reply by return42** [reply]
I think it will not be possible to create a program for this.  The causes for an error pattern are not unique and can be manifold.  I think we will never be able to offer more than *trouble shooting* and *best practice* .. since causes and error treatments will always change, regular maintenance of the wiki is required.  I can not estimate whether there are people in the community who would be willing to maintain such a guide (quick-checks sheet / wiki or whatever).

**Reply by BernieHuang2008** [reply]
Many issues is sent before checking existing issues ... 

Maybe we can map the user's bug report (only generated) to avoid duplicated issues?

**Reply by return42** [reply]
I think it is what we try with ...

![grafik](https://github.com/searxng/searxng/assets/554536/0724c5f0-d558-4e88-aca8-4e87c14db89e)

in the stats page of an engine: https://darmarit.org/searx/stats?engine=duckduckgo

The [Please check for existing bugs about this engine on GitHub](https://github.com/searxng/searxng/issues?q=is%3Aissue+Bug:%20duckduckgo) links to a query of similar bugs.

Unfortunately, there are always people who do not read the instructions and do not follow the link ... but sometimes the openers have a completely different problem than one of the existing issues.  Sometimes the error messages are so different that the simple user would be overwhelmed with his own error messages to assign the existing concrete.

**Reply by BernieHuang2008** [reply]
The search query can contain more information of the error, or there will be mountains of issues with same title, user will be confused. (I had to add more error infos to the query)

Besides, since the search query is more clear, the `submit issue` button can be shown after **clicking** the `check existing issues` link, forcing the user to check existing issues.
(You know better than me, at present, the button is triggered after waiting a couple of seconds )

 _I can do this._


Second, we can use some tools to automatically extract the error&solution from github issue, this can avoid maintaining a check sheet.

**Reply by return42** [reply]
> Besides, since the search query is more clear, the `submit issue` button can be shown after **clicking** the `check existing issues` link, forcing the user to check existing issues.
(You know better than me, at present, the button is triggered after waiting a couple of seconds )

Yes, this is a weak point in the current process ..

>  _I can do this._

If you know of a better solution I would appreciate a PR on this. In that we can then discuss the pros and cons / so far I had no idea we improve it, but we are grateful for any suggestion that represents an improvement :+1:

> Second, we can use some tools to automatically extract the error&solution from github issue, this can avoid maintaining a check sheet.

Not sure what you have in mind ..

**Reply by dalf** [reply]
Just for the records and slightly out of topic:  
* the Engines tab of https://searx.space/ is based on the `/stats/errors` API. For example : https://paulgo.io/stats/errors
* there is an embedded checker
    * it can run on schedule in the webapp. See [settings.yml](https://github.com/searxng/searxng/blob/fd1422a67017443a0dc0773562ec98525b468bde/searx/settings.yml#L240-L250)
    * run from the command line: `make search.checker.bing`.
    * some engine can have some specific tests: see settings.yml, [wikidata engine](https://github.com/searxng/searxng/blob/fd1422a67017443a0dc0773562ec98525b468bde/searx/settings.yml#L598) and [this section](https://github.com/searxng/searxng/blob/fd1422a67017443a0dc0773562ec98525b468bde/searx/settings.yml#L277-L282)

**Reply by unixfox** [reply]
> Or, more Convienent, we can write a script that Auto-checks the problem and provide solution.

That's a good idea! We could have a dedicated page for solutions for the admin. Like we already have /stats

But each solution will point to a wiki page.

Or we could on the same page like https://darmarit.org/searx/stats?engine=duckduckgo, provide an easy link to a wiki page for a solution so that the admin can easily find the solution.

**Reply by BernieHuang2008** [reply]
> But each solution will point to a wiki page.

Do you mean a special section for all the solutions, where each of a solution has a single page in wiki?

**Reply by BernieHuang2008** [reply]
As I said, each problem has a unique `technical report`  (the same you saw on every issue), so we can index the bugs/solutions by their `technical report` or the hash of `technical report`.

we can also use the wiki documents search system.
